Packet++/header/InfiniBandLayer.h

```diff
@@ -0,0 +1,288 @@
+	 * hhh header and get(set) the fff field. The routines named
+	 * hhh_(set_)fff take a packet info struct and find the
+	 * header and field based on the opcode in the packet.
+	 * Conversion to/from network byte order from cpu order is also done.
+	 */
+
+#define RXE_ICRC_SIZE 4
+#define RXE_MAX_HDR_LENGTH 80
+
+	/**
+	 * @struct bth
+	 * Represents an Base Transport Header
+	 */
+#pragma pack(push, 1)
+	struct rxe_bth
+	{
+		uint8_t opcode;
+		uint8_t flags;
+		uint16_t pkey;
+		uint32_t qpn;
+		uint32_t apsn;
+	};
+#pragma pack(pop)
+
+#define BTH_TVER 0x0
+#define BTH_DEF_PKEY 0xffff
+
+#define BTH_SE_MASK 0x80
+#define BTH_MIG_MASK 0x40
+#define BTH_PAD_MASK 0x30
+#define BTH_TVER_MASK 0x0f
+#define BTH_FECN_MASK 0x80000000
+#define BTH_BECN_MASK 0x40000000
+#define BTH_RESV6A_MASK 0x3f000000
+#define BTH_QPN_MASK 0x00ffffff
+#define BTH_ACK_MASK 0x80000000
+#define BTH_RESV7_MASK 0x7f000000
+#define BTH_PSN_MASK 0x00ffffff
+
+	/**
+	 * @class InfiniBandLayer
+	 * Represents an InfiniBand protocol layer
+	 */
+	class InfiniBandLayer : public Layer
+	{
+	public:
+		/**
+		 * A constructor that creates the layer from an existing packet raw data
+		 * @param[in] data A pointer to the raw data (will be casted to bth_header)
+		 * @param[in] dataLen Size of the data in bytes
+		 * @param[in] prevLayer A pointer to the previous layer
+		 * @param[in] packet A pointer to the Packet instance where layer will be stored in
+		 */
+		InfiniBandLayer(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet)
+		    : Layer(data, dataLen, prevLayer, packet, Infiniband)
+		{}
+
+		/**
+		 * A constructor that creates a new rxe_bth header and allocates the data
+		 * @param[in] opcode The operation code
+		 * @param[in] se The solicited event
+		 * @param[in] mig The migration state
+		 * @param[in] pad The pad count
+		 * @param[in] pkey The partition key
+		 * @param[in] qpn The destination queue pair (QP) number
+		 * @param[in] ack_req The acknowledgment request
+		 * @param[in] psn The packet sequence number
+		 */
+		InfiniBandLayer(uint8_t opcode, int se, int mig, int pad, uint16_t pkey, uint32_t qpn, int ack_req,
+		                uint32_t psn);
+
+		/**
+		 * Get a pointer to the BTH header. Notice this points directly to the data, so every change will change
+		 * the actual packet data
+		 * @return A pointer to the bth_header
+		 */
+		rxe_bth* getBthHeader() const
+		{
+			return reinterpret_cast<rxe_bth*>(m_Data);
+		}
+
+		/**
+		 * @return The operation code which defines the interpretation of the remaining header and payload bytes
+		 */
+		uint8_t getOpcode() const;
+
+		/**
+		 * Set operation code
+		 * @param[in] opcode The opcode to set
+		 */
+		void setOpcode(uint8_t opcode) const;
+
+		/**
+		 * @return solicited event that the responder shall invoke the CQ event handler
+		 */
+		uint8_t getSe() const;
+
+		/**
+		 * Set solicited event
+		 * @param[in] se The solicited event to set
+		 */
+		void setSe(int se) const;
```

Comment:
I'd give more explicit names to these getter and setter: `getSoliciteEvent()`, `setSolicitedEvent()`

Same for the other getters and setters, for example:
- `get/setMigrationState`
- `get/setPadCount`
- `get/setTransportHeaderVersion`
- `get/setPartitionKey`
- `get/setQueuePairNumber`
- `get/setPacketSequenceNumber`

Reply:
rename it.